var/spack/repos/builtin/packages/yaml-cpp/package.py

```diff
@@ -32,6 +37,23 @@ class YamlCpp(CMakePackage):
     conflicts('%intel@:11.1', when='@0.6.0:', msg="versions 0.6.0: require c++11 support")
     conflicts('%xl@:13.1', when='@0.6.0:', msg="versions 0.6.0: require c++11 support")
     conflicts('%xl_r@:13.1', when='@0.6.0:', msg="versions 0.6.0: require c++11 support")
+    conflicts('%clang cxxflags="-stdlib=libc++"', when='+tests',
+              msg=yaml_cpp_tests_libcxx_error_msg)
+
+    def flag_handler(self, name, flags):
+        # We cannot catch all conflicts with the conflicts directive because
+        # the user can add arbitrary strings to the flags. Here we can at least
```

Comment:
What do you mean "arbitrary strings"?  Not sure I understand this particular comment.

Reply:
@tgamblin The spec `cxxflags='-O3 -stdlib=libc++'` doesn't satisfy `cxxflags='-stdlib=libc++`, and the user can put in any string that's a valid flag for whichever compiler they're using. So we can't just enumerate the actual conflicts using the `conflicts` directive, so we need to do conflict management in the flag handler.